streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamJoinWindow.java

```diff
@@ -34,9 +34,9 @@
     KStreamJoinWindow(String windowName, long windowSizeMs, long retentionPeriodMs) {
         this.windowName = windowName;
 
-        if (windowSizeMs * 2 > retentionPeriodMs)
+        if (windowSizeMs > retentionPeriodMs)
             throw new TopologyBuilderException("The retention period of the join window "
-                    + windowName + " must at least two times its window size.");
+                    + windowName + " must be no smaller than its window size.");
```

Comment:
Is there a reason why we are using `no smaller` instead of `greater than or equal to`? I generally prefer to avoid logical negations in messages.